index.bs

```diff
@@ -3798,168 +3991,785 @@ To <dfn>get the response data</dfn> given |response|:
+The [=remote end steps=] given |session| and |command parameters| are:
+
+1. Let |blocked requests| be |session|'s [=blocked request map=].
+
+1. Let |request id| be |command parameters|["<code>request</code>"].
+
+1. If |blocked requests| does not [=map/contain=] |request id| then return
+   [=error=] with [=error code=] [=invalid argument=].
+
+   Issue: consider a "<code>no such request</code>" error.
+
+1. Let (<var ignore>request</var>, |phase|, |response|) be |blocked requests|[|request id|].
+
+1. If |phase| is not "<code>authRequired</code>", then return [=error=]
+   with [=error code=] [=invalid argument=].
+
+1. If |command parameters| "<code>action</code> is "<code>cancel</code>", set
+   |response|'s authentication credentials <!--TODO: link--> to
+   "<code>cancelled</code>".
+
+1. If |command parameters| "<code>action</code>" is
+   "<code>provideCredentials</code>":
+
+   1. Let |credentials| be |command parameters|["<code>credentials</code>"].
+
+   1. Asssert: |credentials|{"<code>type</code>"] is "<code>password</code>".
+
+   1. Set |response|'s authentication credentials <!--TODO: link--> to
+      (|credentials|["<code>username</code>"], |credentials|["<code>password</code>"])
+
+1. [=Resume=] with "<code>continue request</code>", |request id|, and
+   (|response|, "<code>incomplete</code>").
+
+1. Return [=success=] with [=data=] null.
+
+</div>
+
+#### The network.failRequest Command ####  {#command-network-failRequest}
+
+The <dfn export for=commands>network.failRequest</dfn> command fails a
+fetch that's blocked by a [=network intercept=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      network.FailRequest = {
+        method: "network.failRequest",
+        params: network.FailRequestParameters
+      }
+
+      network.FailRequestParameters = {
+        request: network.Request,
+      }
+      </pre>
+  </dd>
+</dl>
+
+<div algorithm="remote end steps for network.failRequest">
+The [=remote end steps=] given |session| and |command parameters| are:
+
+1. Let |blocked requests| be |session|'s [=blocked request map=].
+
+1. Let |request id| be |command parameters|["<code>request</code>"].
+
+1. If |blocked requests| does not [=map/contain=] |request id| then return
+   [=error=] with [=error code=] [=invalid argument=].
+
+   Issue: consider a "<code>no such request</code>" error.
+
+1. Let (<var ignore>request</var>, |phase|, <var ignore>response</var>) be
+   |blocked requests|[|request id|].
+
+1. If |phase| is "<code>authRequired</code>", then return [=error=]
+   with [=error code=] [=invalid argument=].
+
+1. Let |response| be a new [=network error=].
+
+1. TODO: Allow setting the precise kind of error
+
+1. [=Resume=] with "<code>continue request</code>", |request id|, and
+   (|response|, "<code>complete</code>").
+
+1. Return [=success=] with data null.
+
+</div>
+
+#### The network.provideResponse Command ####  {#command-network-provideResponse}
```

Comment:
FYI only: CDP calls this [fulfillRequest](https://vanilla.aslushnikov.com/?Fetch.fulfillRequest).

No objections if you prefer to keep is as is though.